builder/comp-builder.nix

```diff
@@ -233,12 +239,15 @@ stdenv.mkDerivation ({
         ${ghc.targetPrefix}ghc-pkg -v0 --package-db ${configFiles}/package.conf.d -f $out/package.conf.d register ${name}.conf
       fi
     ''}
-    ${lib.optionalString (haskellLib.isTest componentId || haskellLib.isBenchmark componentId || haskellLib.isAll componentId) ''
+    ${(lib.optionalString (haskellLib.isTest componentId || haskellLib.isBenchmark componentId || haskellLib.isAll componentId) ''
       mkdir -p $out/${name}
       if [ -f ${testExecutable} ]; then
         cp ${testExecutable} $out/${name}/
       fi
-    ''}
+    '')
+    # In case `setup copy` did not creat this
+    + (lib.optionalString enableSeparateDataOutput "mkdir -p $data")
+    }
```

Comment:
I added this into an existing `${ }` for now so that it does not change the InstallPhase string when `enableSeparateDataOutput` is not set (triggering lots of rebuilds).